src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindVectorProvidersOperation.java

```diff
@@ -0,0 +1,97 @@
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.QueryExecutor;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.PropertyBasedEmbeddingServiceConfig;
+import io.stargate.sgv2.jsonapi.service.operation.model.Operation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Operation that list all available vector providers into the {@link
+ * CommandStatus#EXISTING_VECTOR_PROVIDERS} command status.
+ */
+public record FindVectorProvidersOperation(
+    PropertyBasedEmbeddingServiceConfig propertyBasedEmbeddingServiceConfig) implements Operation {
+  @Override
+  public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
+    return Uni.createFrom()
+        .item(
+            () -> {
+              Map<String, ReturnVectorProviderConfig> vectorProviders = new HashMap<>();
+              if (propertyBasedEmbeddingServiceConfig.vertexai().enabled()) {
+                vectorProviders.put(
+                    "vertexai",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.vertexai())));
+              }
+              if (propertyBasedEmbeddingServiceConfig.huggingface().enabled()) {
+                vectorProviders.put(
+                    "huggingface",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.huggingface())));
+              }
+              if (propertyBasedEmbeddingServiceConfig.openai().enabled()) {
+                vectorProviders.put(
+                    "openai",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.openai())));
+              }
+              if (propertyBasedEmbeddingServiceConfig.cohere().enabled()) {
+                vectorProviders.put(
+                    "cohere",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.cohere())));
+              }
+              if (propertyBasedEmbeddingServiceConfig.nvidia().enabled()) {
+                vectorProviders.put(
+                    "nvidia",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.nvidia())));
+              }
+              return new Result(vectorProviders);
+            });
+  }
+
+  // simple result wrapper
+  private record Result(Map<String, ReturnVectorProviderConfig> vectorProviders)
+      implements Supplier<CommandResult> {
+
+    @Override
+    public CommandResult get() {
+      Map<CommandStatus, Object> statuses =
+          Map.of(CommandStatus.EXISTING_VECTOR_PROVIDERS, vectorProviders);
+      return new CommandResult(statuses);
+    }
+  }
+
+  private record ReturnVectorProviderConfig(
```

Comment:
Why couldn't we use the same objects that we read as provider configuration?

Reply:
As I explained in the previous comment, `parameters` are for external use and `properties` are for internal use. We don't want to expose `properties` so I made a new object here.